### DISCUSSION TITLE
Local File Management Capabilities for Pybricks Code

### DISCUSSION THREAD

**Discussion by thebuilder6** [pybricks/support#1128] [Ideas]
Background: The current implementation of Pybricks as a web application limits users to editing files stored within the browser database. While convenient for quick prototyping or one-off ideas, the lack of native support for managing local files presents significant challenges to larger teams who require more robust collaboration capabilities.

Use Case Scenario: I coach an FLL team of multiple members collaborating on several SPIKE PRIME projects. Each member has installed Pybricks locally on their laptops while accessing Github through separate local repositories to share code snippets or entire projects. 
Currently, using Pybricks with git requires closing all open tabs related to the project in order to prevent it from keeping previous versions in the browser. After closing those tabs, the user would then import the latest version of each file into Pybricks from their local repository. Then, the user edits the file as needed and saves their changes. Next, the student needs to download each edited file to their local repository, which involves confirming whether they want to overwrite the original file and accepting the credibility of code.pybricks.com. Due to the repetitive nature of this process for many files, it significantly affects the workflow and discourages new users from exploring Git. 

Request: Please consider implementing native support for managing local files in Pybricks if reasonable, however any improvement to streamline or automate the local file management process would be helpful.

**Comment by laurensvalk** [comment]
To make things slightly easier, you can download all programs at once as `.zip` file and upload it the next time. This won't really help with the git workflow but it can make the process of handling multiple files a bit less error prone.

Would you say code synchronization across team members (and/or git) is worth having as a premium feature in Pybricks?

For more advanced file management, it may ultimately make more sense for us to develop a Visual Studio Code extension for Pybricks instead of re-inventing these techniques in our web app, but this is not currently on our roadmap.
